File: src/entities/customer.ts
/*    {
        "cid": 1,
        "email": "json@gmail.com",
        "password": "12345",
        "image": "profiles/1001.jpeg",
        "timeCreateId": "2024-11-15T04:59:22.000+00:00",
        "detail": {
            "did": 1,
            "birthday": "2008-11-10T17:00:00.000+00:00",
            "maritalStatus": false,
            "salary": 26000.0,
            "phone": "0645322123"
        }
    }*/
import {Detail} from "./detail.ts";

export class Customer {
    public declare cid: number;
    public declare email: string;
    public declare password: string;
    public declare image: string;
    public declare timeCreateId: Date;
    public declare detail : Detail

}

File: src/main/java/com/ttknp/onetoonehasrelationtableandsecurity/entities/Customer.java
package com.ttknp.onetoonehasrelationtableandsecurity.entities;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.util.Date;

@Entity
@Table(name = "customers")
@Data
@NoArgsConstructor
@ToString
public class Customer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long cid;
    private String email;
    private String password;
    private String image;
    private Date timeCreateId; // Date type can map timestamp type in mysql
    // I have relation table this way below for join it
    @OneToOne(cascade = CascadeType.ALL)
    @JoinTable(
            name = "customers_details",
            joinColumns = {
                    // mean to access customers.cid and referent customers_details.cid
                    @JoinColumn(name = "cid", referencedColumnName = "cid")
            },
            inverseJoinColumns = {
                    @JoinColumn(name = "did", referencedColumnName = "did")
            })
    private Detail detail;
}
